Sources/TuistAutomation/Device/DeviceController.swift

```diff
@@ -162,7 +162,7 @@ extension PhysicalDevice {
             id: device.hardwareProperties.udid,
             name: device.deviceProperties.name,
             platform: platform,
-            osVersion: device.deviceProperties.osVersionNumber
+            osVersion: device.deviceProperties.osVersionNumber ?? "Unknown"
```

Comment:
I'd keep this `nil` and make also the `osVersion` of the `PhysicalDevice` optional. I'd let the consumer of `PhysicalDevice` decide what to do when the value is `nil`. Instead of showing `Unknown`, I'd just skip showing the label altogether.

Reply:
Can we also test this scenario in `DeviceControllerTests.swift`?

Reply:
Thanks for the review!

Should we temporarily throw a `deviceUnreachable` error and alert the user to reconnect the device when attempting to select an unreachable device?

Reply:
I have added the test case. Please review it again.

Reply:
> Should we temporarily throw a deviceUnreachable error and alert the user to reconnect the device when attempting to select an unreachable device?

I'd either:
- not show such a device at all
- show the device with a message that it is unreachable. Make the row disabled and add a button to try and reload the device.

Reply:
Alright, I’ll go ahead and hide the unreachable devices for now. [`8f94c30` (#6864)](https://github.com/tuist/tuist/pull/6864/commits/8f94c30cc5f0404038397e57a013d0fdb4137e6d)